src/core/frames/frame_renderer.js

```diff
@@ -15,6 +24,15 @@ export class FrameRenderer extends Renderer {
     }
   }
 
+  static morph(currentElement, newElement) {
+    dispatch("turbo:before-frame-morph", {
+      target: currentElement,
+      detail: { currentElement, newElement }
+    })
+
+    Morph.render(currentElement, newElement, "innerHTML")
```

Comment:
I've just found that Idiomorph's `innerHTML` mode doesn't work as we expect: it only consider this option for the target node, not the new element; so, the correct code would be:

```suggestion
    Morph.render(currentElement, newElement.children, "innerHTML")
```